classes/template/PKPTemplateManager.php

```diff
@@ -2504,6 +2507,57 @@ public function smartyLoadNavigationMenuArea($params, $smarty = null)
         return $this->fetch($menuTemplatePath);
     }
 
+    /**
+     * Smarty usage: {include_blade file=$file params1=$params1 params2=$params2 ...}
+     *
+     * Custom Smarty function for injecting a blade view in a smarty template
+     *
+     * @param array $params associative array
+     * @param Smarty|null $smarty
+     *
+     * @return string The compiled content of the blade view
+     * 
+     * @throws Exception If the file parameter is missing or the blade view does not exist  
+     */
+    public function smartyIncludeBlade($params, $smarty = null): string
+    {
+        if (!isset($params['file'])) {
+            throw new Exception('file parameter is missing in {include_blade}');
+        }
+
+        $file = $params['file'];
+
+        if (str_contains($file, '..') || str_contains($file, '\\') || str_starts_with($file, '/')) {
+            throw new Exception("Invalid file path: path traversal or absolute paths not allowed in given blade file {$file}");
+        }
+
+        $file = str_replace(['/', '.blade.php', '.blade'], ['.', '', ''], $file);
```

Comment:
We shouldn't be coding for `.blade.php` files anywhere; they should always be just `.blade` (at least until we move the codebase outside the web root).